llama_index/llms/cohere_utils.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Type, Union
+
+import cohere
+from cohere.client import Client
+from cohere.client_async import AsyncClient
+from cohere.responses import Chat, Generations
```

Comment:
I _think_ this introduces a hard dependency on the cohere client library, which we'd prefer to avoid.

Ideally we can wrap this with try/catch and only import the library when user intends to use Cohere, not on import.

Reply:
The only barrier to this was the completion_with_retry and acompletion_with_retry functions required a cohere.client type in the function arguements annotation for the client parameter. I'll change the annotation to be Any type if thats acceptable